upup/pkg/fi/cloudup/awstasks/ebsvolume.go

```diff
@@ -181,6 +178,29 @@ func (_ *EBSVolume) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *EBSVolume) e
 			}
 		}
 
+		describeInput := &ec2.DescribeVolumesInput{
+			VolumeIds: []*string{
+				a.ID,
+			},
+		}
+
+		// Validate encrypted volume.
+		// Default encryption is configured in AWS, so even if users change encryptedVolume,
+		// it is not a problem as long as it matches the status on AWS.
+		volumes, err := t.Cloud.EC2().DescribeVolumes(describeInput)
+		if err != nil {
+			return fmt.Errorf("error get current volume: %v", err)
+		}
+		if len(volumes.Volumes) == 0 {
+			return fmt.Errorf("could not find volume for: %s", *a.ID)
+		}
+		currentVolume := volumes.Volumes[0]
+		if changes.Encrypted != nil {
+			if *e.Encrypted != *currentVolume.Encrypted {
+				return fmt.Errorf("encryptedVolume cannot be changed")
+			}
+		}
+
```

Comment:
This seems to be doing the same thing changes.Encrypted does, by comparing actual and expected.